File: Benchmarking/src/gen_12_28.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: s0
    var p1: s0
  }
  @inline(never)
  func f0(_ v0: s1, _ v1: Double) -> Double {
    let v4: s0 = v0.p1
    let v2: [s0] = [v4, v4, v4, v4, v4]
    let v6: s0 = v2[1]
    let v3: s0 = v2[0]
    let v7: s0 = v0.p1
    let v9: [[Double]] = v3.p1
    let v19: [[Double]] = v6.p1
    let v31: [Double] = v19[1]
    let v15: Double = v31[0]
    let v45: [Double] = v9[2]
    let v26: Double = v15 - v15
    let v40: [[Double]] = v7.p0
    let v51: [Double] = v40[0]
    var v36: [Double] = v31
    v36[0] = v26
    let v68: Double = v51[0]
    v36[0] = v68
    var v125: Double = v15
    var v72: Double = v125
    var v58: [Double] = v45
    let v80: Double = v72 - v15
    var v59: [Double] = v31
    let v91: [[Double]] = [v58, v45, v45, v36]
    v36 = v59
    let v210: [Double] = v91[2]
    var v123: [Double] = v210
    v58[0] = v80
    let v81: Double = v123[0]
    return v81
  }
  func benchmark() {
    let v0: s1 = s1(p0: s0(p0: [[0.0], [1.0], [2.0]], p1: [[3.0], [4.0], [5.0]]), p1: s0(p0: [[6.0], [7.0], [8.0]], p1: [[9.0], [10.0], [11.0]]))
    let v1: Double = 12.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 12.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen_12_28.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  struct s1 {
    s0 p0;
    s0 p1;
    s1(s0 v0, s0 v1): p0(v0), p1(v1) { }
  };
  __attribute__((noinline))
  double f0(const s1 &v0, const double &v1) {
    const s0 v4 = v0.p1;
    const std::vector<s0> v2 { v4, v4, v4, v4, v4 };
    const s0 v6 = v2[1];
    const s0 v3 = v2[0];
    const s0 v7 = v0.p1;
    const std::vector<std::vector<double>> v9 = v3.p1;
    const std::vector<std::vector<double>> v19 = v6.p1;
    const std::vector<double> v31 = v19[1];
    const double v15 = v31[0];
    const std::vector<double> v45 = v9[2];
    const double v26 = v15 - v15;
    const std::vector<std::vector<double>> v40 = v7.p0;
    const std::vector<double> v51 = v40[0];
    std::vector<double> v36 = v31;
    v36[0] = v26;
    const double v68 = v51[0];
    v36[0] = v68;
    double v125 = v15;
    double v72 = v125;
    std::vector<double> v58 = v45;
    const double v80 = v72 - v15;
    std::vector<double> v59 = v31;
    const std::vector<std::vector<double>> v91 { v58, v45, v45, v36 };
    v36 = v59;
    const std::vector<double> v210 = v91[2];
    std::vector<double> v123 = v210;
    v58[0] = v80;
    const double v81 = v123[0];
    return v81;
  }
  int main() {
    s1 v0({ { { { 0.0 }, { 1.0 }, { 2.0 } }, { { 3.0 }, { 4.0 }, { 5.0 } } }, { { { 6.0 }, { 7.0 }, { 8.0 } }, { { 9.0 }, { 10.0 }, { 11.0 } } } });
    double v1(12.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }
